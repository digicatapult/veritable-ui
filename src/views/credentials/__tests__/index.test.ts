import { expect } from 'chai'
import { describe, it } from 'mocha'

import { default as CredentialsTemplates } from '../index.js'
import { AliceCredentials, BobCredentials } from './fixtures.js'

describe('CredentialsTemplates', () => {
  describe('listPage', () => {
    it('renders with no credentials', async () => {
      const templates = new CredentialsTemplates()
      const rendered = await templates.listPage([])
      expect(rendered).to.matchSnapshot()
    })

    it("renders alice's credentials", async () => {
      const templates = new CredentialsTemplates()
      const rendered = await templates.listPage(AliceCredentials)
      expect(rendered).to.matchSnapshot()
    })

    it("renders bob's credentials", async () => {
      const templates = new CredentialsTemplates()
      const rendered = await templates.listPage(BobCredentials)
      expect(rendered).to.matchSnapshot()
    })

    describe('credentials case insensitive searches', () => {
      it('filters based on lower case input', async () => {
        const templates = new CredentialsTemplates()
        const rendered = await templates.listPage(AliceCredentials, 'digi')

        expect(rendered.includes('DIGITAL CATAPULT')).to.equal(true)
        expect(rendered).to.matchSnapshot()
      })

      it('filters based on upper case input', async () => {
        const templates = new CredentialsTemplates()
        const rendered = await templates.listPage(AliceCredentials, 'CARE')

        expect(rendered.includes('CARE')).to.equal(true)
        expect(rendered).to.matchSnapshot()
      })
    })
  })
})
