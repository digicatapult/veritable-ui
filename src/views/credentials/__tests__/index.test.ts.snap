// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CredentialsTemplates listPage credentials case insensitive searches filters based on lower case input 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"/lib/htmx.org/htmx.min.js\\"></script><script src=\\"/lib/htmx-ext-json-enc/json-enc.js\\"></script><script src=\\"/public/scripts/auth-redirect.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" href=\\"/public/images/favicon-dark.ico\\" media=\\"(prefers-color-scheme: dark)\\"/><link rel=\\"icon\\" type=\\"image/ico\\" href=\\"/public/images/favicon.ico\\" media=\\"(prefers-color-scheme: light), (prefers-color-scheme: no-preference)\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/query.css\\"/><title>Veritable - Credentials</title></head><body hx-ext=\\"json-enc\\"><nav id=\\"side-bar\\"><img src=\\"/public/images/logo-square.svg\\"/><div class=\\"profile-icon\\" style=\\"background-color:#38b6ff;\\">A</div><a title=\\"categories\\" href=\\"/\\" style=\\"--background-image:url('/public/images/category.svg');\\"></a><a title=\\"queries\\" href=\\"/queries\\" style=\\"--background-image:url('/public/images/query.svg');\\"></a><a title=\\"credentials\\" href=\\"/credentials\\" data-active style=\\"--background-image:url('/public/images/certification.svg');\\"></a><a title=\\"connections\\" href=\\"/connection\\" style=\\"--background-image:url('/public/images/connection.svg');\\"></a><a title=\\"settings\\" href=\\"#\\" class=\\"disabled\\" style=\\"--background-image:url('/public/images/setting.svg');\\"></a></nav><main><div id=\\"content-header\\"><h1>Credential Management</h1><div id=\\"content-header-nav\\"><a title=\\"home\\" class=\\"nav icon\\" href=\\"/\\"></a><a title=\\"Credential Management\\" href=\\"/credentials\\">Credential Management</a></div></div><div id=\\"content-main\\" hx-get=\\"/credentials\\" hx-trigger=\\"every 10s\\" hx-select=\\"#search-results\\" hx-target=\\"#search-results\\" hx-swap=\\"outerHTML\\" hx-include=\\"#queries-search-input\\"><div class=\\"list-page-header\\"><span>Credential Management</span><a class=\\"button disabled\\" href=\\"#\\" data-variant=\\"filled\\"><div style=\\"--button-icon:url(&#34;/public/images/plus.svg&#34;);\\"></div><span>New Credential</span></a></div><div class=\\"list-page \\"><div class=\\"list-nav\\"><span>Credentials</span><input id=\\"queries-search-input\\" class=\\"search-window\\" type=\\"search\\" name=\\"search\\" value=\\"digi\\" placeholder=\\"Search\\" hx-get=\\"/credentials\\" hx-trigger=\\"input changed delay:500ms, search\\" hx-target=\\"#search-results\\" hx-select=\\"#search-results\\" hx-swap=\\"outerHTML\\"/></div><table class=\\"list-page\\"><thead><th><span>Company Name</span><a class=\\"list-filter-icon icon disabled\\"></a></th><th><span>Credential Type</span><a class=\\"list-filter-icon icon disabled\\"></a></th><th><span>Direction</span><a class=\\"list-filter-icon icon disabled\\"></a></th><th><span>Crendetial status</span><a class=\\"list-filter-icon icon disabled\\"></a></th><th><span>Actions</span><a class=\\"list-filter-icon icon disabled\\"></a></th></thead><tbody id=\\"search-results\\"><tr><td>DIGITAL CATAPULT</td><td>Supplier credentials</td><td><p>Given</p></td><td><div class=\\"list-item-status\\" data-status=\\"success\\">Completed</div></td><td><a class=\\"button disabled\\" href=\\"#\\" data-variant=\\"outlined\\"><div style=\\"--button-icon:url(&#34;/public/images/dot-icon.svg&#34;);\\"></div><span>View Credential</span></a></td></tr><tr><td>CARE ONUS LTD</td><td>Supplier credentials</td><td><p>Received</p></td><td><div class=\\"list-item-status\\" data-status=\\"success\\">Completed</div></td><td><a class=\\"button disabled\\" href=\\"#\\" data-variant=\\"outlined\\"><div style=\\"--button-icon:url(&#34;/public/images/dot-icon.svg&#34;);\\"></div><span>View Credential</span></a></td></tr></tbody></table></div></div></main></body></html>"`;

exports[`CredentialsTemplates listPage credentials case insensitive searches filters based on upper case input 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"/lib/htmx.org/htmx.min.js\\"></script><script src=\\"/lib/htmx-ext-json-enc/json-enc.js\\"></script><script src=\\"/public/scripts/auth-redirect.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" href=\\"/public/images/favicon-dark.ico\\" media=\\"(prefers-color-scheme: dark)\\"/><link rel=\\"icon\\" type=\\"image/ico\\" href=\\"/public/images/favicon.ico\\" media=\\"(prefers-color-scheme: light), (prefers-color-scheme: no-preference)\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/query.css\\"/><title>Veritable - Credentials</title></head><body hx-ext=\\"json-enc\\"><nav id=\\"side-bar\\"><img src=\\"/public/images/logo-square.svg\\"/><div class=\\"profile-icon\\" style=\\"background-color:#38b6ff;\\">A</div><a title=\\"categories\\" href=\\"/\\" style=\\"--background-image:url('/public/images/category.svg');\\"></a><a title=\\"queries\\" href=\\"/queries\\" style=\\"--background-image:url('/public/images/query.svg');\\"></a><a title=\\"credentials\\" href=\\"/credentials\\" data-active style=\\"--background-image:url('/public/images/certification.svg');\\"></a><a title=\\"connections\\" href=\\"/connection\\" style=\\"--background-image:url('/public/images/connection.svg');\\"></a><a title=\\"settings\\" href=\\"#\\" class=\\"disabled\\" style=\\"--background-image:url('/public/images/setting.svg');\\"></a></nav><main><div id=\\"content-header\\"><h1>Credential Management</h1><div id=\\"content-header-nav\\"><a title=\\"home\\" class=\\"nav icon\\" href=\\"/\\"></a><a title=\\"Credential Management\\" href=\\"/credentials\\">Credential Management</a></div></div><div id=\\"content-main\\" hx-get=\\"/credentials\\" hx-trigger=\\"every 10s\\" hx-select=\\"#search-results\\" hx-target=\\"#search-results\\" hx-swap=\\"outerHTML\\" hx-include=\\"#queries-search-input\\"><div class=\\"list-page-header\\"><span>Credential Management</span><a class=\\"button disabled\\" href=\\"#\\" data-variant=\\"filled\\"><div style=\\"--button-icon:url(&#34;/public/images/plus.svg&#34;);\\"></div><span>New Credential</span></a></div><div class=\\"list-page \\"><div class=\\"list-nav\\"><span>Credentials</span><input id=\\"queries-search-input\\" class=\\"search-window\\" type=\\"search\\" name=\\"search\\" value=\\"CARE\\" placeholder=\\"Search\\" hx-get=\\"/credentials\\" hx-trigger=\\"input changed delay:500ms, search\\" hx-target=\\"#search-results\\" hx-select=\\"#search-results\\" hx-swap=\\"outerHTML\\"/></div><table class=\\"list-page\\"><thead><th><span>Company Name</span><a class=\\"list-filter-icon icon disabled\\"></a></th><th><span>Credential Type</span><a class=\\"list-filter-icon icon disabled\\"></a></th><th><span>Direction</span><a class=\\"list-filter-icon icon disabled\\"></a></th><th><span>Crendetial status</span><a class=\\"list-filter-icon icon disabled\\"></a></th><th><span>Actions</span><a class=\\"list-filter-icon icon disabled\\"></a></th></thead><tbody id=\\"search-results\\"><tr><td>DIGITAL CATAPULT</td><td>Supplier credentials</td><td><p>Given</p></td><td><div class=\\"list-item-status\\" data-status=\\"success\\">Completed</div></td><td><a class=\\"button disabled\\" href=\\"#\\" data-variant=\\"outlined\\"><div style=\\"--button-icon:url(&#34;/public/images/dot-icon.svg&#34;);\\"></div><span>View Credential</span></a></td></tr><tr><td>CARE ONUS LTD</td><td>Supplier credentials</td><td><p>Received</p></td><td><div class=\\"list-item-status\\" data-status=\\"success\\">Completed</div></td><td><a class=\\"button disabled\\" href=\\"#\\" data-variant=\\"outlined\\"><div style=\\"--button-icon:url(&#34;/public/images/dot-icon.svg&#34;);\\"></div><span>View Credential</span></a></td></tr></tbody></table></div></div></main></body></html>"`;

exports[`CredentialsTemplates listPage renders alice's credentials 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"/lib/htmx.org/htmx.min.js\\"></script><script src=\\"/lib/htmx-ext-json-enc/json-enc.js\\"></script><script src=\\"/public/scripts/auth-redirect.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" href=\\"/public/images/favicon-dark.ico\\" media=\\"(prefers-color-scheme: dark)\\"/><link rel=\\"icon\\" type=\\"image/ico\\" href=\\"/public/images/favicon.ico\\" media=\\"(prefers-color-scheme: light), (prefers-color-scheme: no-preference)\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/query.css\\"/><title>Veritable - Credentials</title></head><body hx-ext=\\"json-enc\\"><nav id=\\"side-bar\\"><img src=\\"/public/images/logo-square.svg\\"/><div class=\\"profile-icon\\" style=\\"background-color:#38b6ff;\\">A</div><a title=\\"categories\\" href=\\"/\\" style=\\"--background-image:url('/public/images/category.svg');\\"></a><a title=\\"queries\\" href=\\"/queries\\" style=\\"--background-image:url('/public/images/query.svg');\\"></a><a title=\\"credentials\\" href=\\"/credentials\\" data-active style=\\"--background-image:url('/public/images/certification.svg');\\"></a><a title=\\"connections\\" href=\\"/connection\\" style=\\"--background-image:url('/public/images/connection.svg');\\"></a><a title=\\"settings\\" href=\\"#\\" class=\\"disabled\\" style=\\"--background-image:url('/public/images/setting.svg');\\"></a></nav><main><div id=\\"content-header\\"><h1>Credential Management</h1><div id=\\"content-header-nav\\"><a title=\\"home\\" class=\\"nav icon\\" href=\\"/\\"></a><a title=\\"Credential Management\\" href=\\"/credentials\\">Credential Management</a></div></div><div id=\\"content-main\\" hx-get=\\"/credentials\\" hx-trigger=\\"every 10s\\" hx-select=\\"#search-results\\" hx-target=\\"#search-results\\" hx-swap=\\"outerHTML\\" hx-include=\\"#queries-search-input\\"><div class=\\"list-page-header\\"><span>Credential Management</span><a class=\\"button disabled\\" href=\\"#\\" data-variant=\\"filled\\"><div style=\\"--button-icon:url(&#34;/public/images/plus.svg&#34;);\\"></div><span>New Credential</span></a></div><div class=\\"list-page \\"><div class=\\"list-nav\\"><span>Credentials</span><input id=\\"queries-search-input\\" class=\\"search-window\\" type=\\"search\\" name=\\"search\\" value=\\"\\" placeholder=\\"Search\\" hx-get=\\"/credentials\\" hx-trigger=\\"input changed delay:500ms, search\\" hx-target=\\"#search-results\\" hx-select=\\"#search-results\\" hx-swap=\\"outerHTML\\"/></div><table class=\\"list-page\\"><thead><th><span>Company Name</span><a class=\\"list-filter-icon icon disabled\\"></a></th><th><span>Credential Type</span><a class=\\"list-filter-icon icon disabled\\"></a></th><th><span>Direction</span><a class=\\"list-filter-icon icon disabled\\"></a></th><th><span>Crendetial status</span><a class=\\"list-filter-icon icon disabled\\"></a></th><th><span>Actions</span><a class=\\"list-filter-icon icon disabled\\"></a></th></thead><tbody id=\\"search-results\\"><tr><td>DIGITAL CATAPULT</td><td>Supplier credentials</td><td><p>Given</p></td><td><div class=\\"list-item-status\\" data-status=\\"success\\">Completed</div></td><td><a class=\\"button disabled\\" href=\\"#\\" data-variant=\\"outlined\\"><div style=\\"--button-icon:url(&#34;/public/images/dot-icon.svg&#34;);\\"></div><span>View Credential</span></a></td></tr><tr><td>CARE ONUS LTD</td><td>Supplier credentials</td><td><p>Received</p></td><td><div class=\\"list-item-status\\" data-status=\\"success\\">Completed</div></td><td><a class=\\"button disabled\\" href=\\"#\\" data-variant=\\"outlined\\"><div style=\\"--button-icon:url(&#34;/public/images/dot-icon.svg&#34;);\\"></div><span>View Credential</span></a></td></tr></tbody></table></div></div></main></body></html>"`;

exports[`CredentialsTemplates listPage renders bob's credentials 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"/lib/htmx.org/htmx.min.js\\"></script><script src=\\"/lib/htmx-ext-json-enc/json-enc.js\\"></script><script src=\\"/public/scripts/auth-redirect.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" href=\\"/public/images/favicon-dark.ico\\" media=\\"(prefers-color-scheme: dark)\\"/><link rel=\\"icon\\" type=\\"image/ico\\" href=\\"/public/images/favicon.ico\\" media=\\"(prefers-color-scheme: light), (prefers-color-scheme: no-preference)\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/query.css\\"/><title>Veritable - Credentials</title></head><body hx-ext=\\"json-enc\\"><nav id=\\"side-bar\\"><img src=\\"/public/images/logo-square.svg\\"/><div class=\\"profile-icon\\" style=\\"background-color:#38b6ff;\\">A</div><a title=\\"categories\\" href=\\"/\\" style=\\"--background-image:url('/public/images/category.svg');\\"></a><a title=\\"queries\\" href=\\"/queries\\" style=\\"--background-image:url('/public/images/query.svg');\\"></a><a title=\\"credentials\\" href=\\"/credentials\\" data-active style=\\"--background-image:url('/public/images/certification.svg');\\"></a><a title=\\"connections\\" href=\\"/connection\\" style=\\"--background-image:url('/public/images/connection.svg');\\"></a><a title=\\"settings\\" href=\\"#\\" class=\\"disabled\\" style=\\"--background-image:url('/public/images/setting.svg');\\"></a></nav><main><div id=\\"content-header\\"><h1>Credential Management</h1><div id=\\"content-header-nav\\"><a title=\\"home\\" class=\\"nav icon\\" href=\\"/\\"></a><a title=\\"Credential Management\\" href=\\"/credentials\\">Credential Management</a></div></div><div id=\\"content-main\\" hx-get=\\"/credentials\\" hx-trigger=\\"every 10s\\" hx-select=\\"#search-results\\" hx-target=\\"#search-results\\" hx-swap=\\"outerHTML\\" hx-include=\\"#queries-search-input\\"><div class=\\"list-page-header\\"><span>Credential Management</span><a class=\\"button disabled\\" href=\\"#\\" data-variant=\\"filled\\"><div style=\\"--button-icon:url(&#34;/public/images/plus.svg&#34;);\\"></div><span>New Credential</span></a></div><div class=\\"list-page \\"><div class=\\"list-nav\\"><span>Credentials</span><input id=\\"queries-search-input\\" class=\\"search-window\\" type=\\"search\\" name=\\"search\\" value=\\"\\" placeholder=\\"Search\\" hx-get=\\"/credentials\\" hx-trigger=\\"input changed delay:500ms, search\\" hx-target=\\"#search-results\\" hx-select=\\"#search-results\\" hx-swap=\\"outerHTML\\"/></div><table class=\\"list-page\\"><thead><th><span>Company Name</span><a class=\\"list-filter-icon icon disabled\\"></a></th><th><span>Credential Type</span><a class=\\"list-filter-icon icon disabled\\"></a></th><th><span>Direction</span><a class=\\"list-filter-icon icon disabled\\"></a></th><th><span>Crendetial status</span><a class=\\"list-filter-icon icon disabled\\"></a></th><th><span>Actions</span><a class=\\"list-filter-icon icon disabled\\"></a></th></thead><tbody id=\\"search-results\\"><tr><td>CARE ONUS LTD</td><td>Supplier credentials</td><td><p>Given</p></td><td><div class=\\"list-item-status\\" data-status=\\"success\\">Completed</div></td><td><a class=\\"button disabled\\" href=\\"#\\" data-variant=\\"outlined\\"><div style=\\"--button-icon:url(&#34;/public/images/dot-icon.svg&#34;);\\"></div><span>View Credential</span></a></td></tr><tr><td>DIGITAL CATAPULT</td><td>Supplier credentials</td><td><p>Received</p></td><td><div class=\\"list-item-status\\" data-status=\\"success\\">Completed</div></td><td><a class=\\"button disabled\\" href=\\"#\\" data-variant=\\"outlined\\"><div style=\\"--button-icon:url(&#34;/public/images/dot-icon.svg&#34;);\\"></div><span>View Credential</span></a></td></tr></tbody></table></div></div></main></body></html>"`;

exports[`CredentialsTemplates listPage renders with no credentials 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"/lib/htmx.org/htmx.min.js\\"></script><script src=\\"/lib/htmx-ext-json-enc/json-enc.js\\"></script><script src=\\"/public/scripts/auth-redirect.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" href=\\"/public/images/favicon-dark.ico\\" media=\\"(prefers-color-scheme: dark)\\"/><link rel=\\"icon\\" type=\\"image/ico\\" href=\\"/public/images/favicon.ico\\" media=\\"(prefers-color-scheme: light), (prefers-color-scheme: no-preference)\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/query.css\\"/><title>Veritable - Credentials</title></head><body hx-ext=\\"json-enc\\"><nav id=\\"side-bar\\"><img src=\\"/public/images/logo-square.svg\\"/><div class=\\"profile-icon\\" style=\\"background-color:#38b6ff;\\">A</div><a title=\\"categories\\" href=\\"/\\" style=\\"--background-image:url('/public/images/category.svg');\\"></a><a title=\\"queries\\" href=\\"/queries\\" style=\\"--background-image:url('/public/images/query.svg');\\"></a><a title=\\"credentials\\" href=\\"/credentials\\" data-active style=\\"--background-image:url('/public/images/certification.svg');\\"></a><a title=\\"connections\\" href=\\"/connection\\" style=\\"--background-image:url('/public/images/connection.svg');\\"></a><a title=\\"settings\\" href=\\"#\\" class=\\"disabled\\" style=\\"--background-image:url('/public/images/setting.svg');\\"></a></nav><main><div id=\\"content-header\\"><h1>Credential Management</h1><div id=\\"content-header-nav\\"><a title=\\"home\\" class=\\"nav icon\\" href=\\"/\\"></a><a title=\\"Credential Management\\" href=\\"/credentials\\">Credential Management</a></div></div><div id=\\"content-main\\" hx-get=\\"/credentials\\" hx-trigger=\\"every 10s\\" hx-select=\\"#search-results\\" hx-target=\\"#search-results\\" hx-swap=\\"outerHTML\\" hx-include=\\"#queries-search-input\\"><div class=\\"list-page-header\\"><span>Credential Management</span><a class=\\"button disabled\\" href=\\"#\\" data-variant=\\"filled\\"><div style=\\"--button-icon:url(&#34;/public/images/plus.svg&#34;);\\"></div><span>New Credential</span></a></div><div class=\\"list-page \\"><div class=\\"list-nav\\"><span>Credentials</span><input id=\\"queries-search-input\\" class=\\"search-window\\" type=\\"search\\" name=\\"search\\" value=\\"\\" placeholder=\\"Search\\" hx-get=\\"/credentials\\" hx-trigger=\\"input changed delay:500ms, search\\" hx-target=\\"#search-results\\" hx-select=\\"#search-results\\" hx-swap=\\"outerHTML\\"/></div><table class=\\"list-page\\"><thead><th><span>Company Name</span><a class=\\"list-filter-icon icon disabled\\"></a></th><th><span>Credential Type</span><a class=\\"list-filter-icon icon disabled\\"></a></th><th><span>Direction</span><a class=\\"list-filter-icon icon disabled\\"></a></th><th><span>Crendetial status</span><a class=\\"list-filter-icon icon disabled\\"></a></th><th><span>Actions</span><a class=\\"list-filter-icon icon disabled\\"></a></th></thead><tbody id=\\"search-results\\"><tr><td>No Credentials found</td></tr></tbody></table></div></div></main></body></html>"`;
